### ISSUE TITLE
Error Message: App generation failed with exit code 1

### ISSUE THREAD

**Issue by trimline-gaiustemple** [microsoft/AL-Go#764]
Hello,

I am getting the following error:
Error: Unexpected error when running action. Error Message: App generation failed with exit code 1, StackTrace: at <ScriptBlock>, C:\ProgramData\BcContainerHelper\6.0.0\BcContainerHelper\AppHandling\Compile-AppInNavContainer.ps1: line 743 <- at Compile-AppInBcContainer, C:\ProgramData\BcContainerHelper\6.0.0\BcContainerHelper\AppHandling\Compile-AppInNavContainer.ps1: line 743 <- at <ScriptBlock>

The app compiles fine using both a V22 and V23 online sandbox, it just seems to be in the GitHub actions I am getting this error.  Am I doing something wrong?

**Comment by freddydk** [comment]
Please supply the full log of the compile step, that might help us help you.

**Comment by trimline-gaiustemple** [comment]
Hello @freddydk 

Thanks for the quick reply.
<img width="1559" alt="image" src="https://github.com/microsoft/AL-Go/assets/102605083/8cee6fc9-7d80-47d1-98df-1fa132a007f7">

Does that show enough information

**Comment by freddydk** [comment]
Nope, you would need to go to the top hand corner, download the logs and attach them - or copy paste the entire Run Pipeline (expanded) here

**Comment by trimline-gaiustemple** [comment]
Ah - I have found the issue.  Rookie mistake, before when I was looking into it I expanded the section directly above the error but not the second to last 🤦‍♂️

The error was `Error: AL0270 The control '"Promised Delivery Date"' is not found in the target 'Posted Sales Shipment - Update'`

I was extending the "Posted Sales Shipment - Update" page, using `addafter("Promised Delivery Date")`.  Either this field is removed by default on that page in v23 or an extension in our env has added `"Promised Delivery Date"` to the page (although there are no dependencies set, and it compliles ok locally without any set).

Apologies, I know what to look out for next time!
